warehouse/oidc/models/github.py

```diff
@@ -356,13 +359,18 @@ def verify_url(self, url: str):
         The suffix `.git` in repo URLs is ignored, since `github.com/org/repo.git`
         always redirects to `github.com/org/repo`. This does not apply to subpaths,
         like `github.com/org/repo.git/issues`, which do not redirect to the correct URL.
+
+        GitHub uses case-insensitive owner/repo slugs - so we perform a case-insensitive
+        comparison.
         """
+        url = url.lower()
```

Comment:
FLAG: This is a (small) over-approximation because `https://github.com/pypi/warehouse.GIT` would also be verified, while the URL cannot be resolved on GitHub.

Reply:
I don't think we need to call `url.lower()` here, since `super().verify_url(url)` will also call `lower()` on the URL. Removing that call would also mean that we won't verify invalid URLs ending in `.GIT`.

I think adding the `lower()` call below, in 
```python
        user_uri = rfc3986.api.uri_reference(url.lower()).normalize()
```

should work

Reply:
Fixed in the new version